### ISSUE TITLE
Multi output model history validation accuracy problem.

### ISSUE THREAD

**Issue by tuturis** [tensorflow/tfjs#5941]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Windows 10,  and whatever codesandox.io runs on
- TensorFlow.js installed from (npm or script link): npm @tensorflow/tfjs-node@3.12.0 and @tensorflow/tfjs-node-gpu@3.12.0

**Describe the current behavior**
`fitDataset` of `tf.model` with multiple (3) outputs show invalid history entry for validation accuracy of 3rd output, on both `tfjs-node` and `tfjs-node-gpu`.
Also there's an type error when trying to pass output layers into array of model outputs.

**Describe the expected behavior**
Show proper metrics for outputs 

**Standalone code to reproduce the issue**
code to reproduce 
```
import {
  layers,
  train,
  model,
  SymbolicTensor,
  tensor2d,
  tensor1d,
  data
} from "@tensorflow/tfjs-node";
const { flatten, dense, input, conv1d, maxPooling1d } = layers;
(async () => {
  const inputs = input({
    shape: [8, 4],
    name: "input331"
  });

  const conv1D0 = conv1d({ filters: 32, kernelSize: 4 }).apply(inputs);
  const maxPooling1D0 = maxPooling1d({ poolSize: 4, padding: "valid" }).apply(
    conv1D0
  );
  const flatten0 = flatten().apply(maxPooling1D0);
  const dense00 = dense({ units: 512, activation: "relu" }).apply(flatten0);
  const dense10 = dense({ units: 512, activation: "relu" }).apply(flatten0);
  const dense20 = dense({ units: 512, activation: "relu" }).apply(flatten0);

  const dense01 = dense({ units: 256, activation: "relu" }).apply(dense00);
  const dense11 = dense({ units: 256, activation: "relu" }).apply(dense10);
  const dense21 = dense({ units: 256, activation: "relu" }).apply(dense20);

  const y1 = dense({ units: 1, name: "y1" }).apply(dense01);
  const y2 = dense({ units: 1, name: "y2" }).apply(dense11);
  const y3 = dense({ units: 1, name: "y3" }).apply(dense21);

  let tfModel = model({
    inputs: inputs,
    // @ts-ignore here is also a bug with types...
    outputs: [y1, y2, y3],
    name: "1"
  });

  tfModel.compile({
    optimizer: train.adam(0.0001),
    loss: {
      y1: "meanSquaredError",
      y2: "meanSquaredError",
      y3: "meanSquaredError"
    },
    metrics: {
      y1: "accuracy",
      y2: "accuracy",
      y3: "accuracy"
    }
  });

  tfModel.summary();

  const array = [
    [2, 3, 1, 2],
    [2, 4, 2, 3],
    [3, 5, 2, 4],
    [4, 5, 3, 3],
    [3, 6, 3, 5],
    [5, 7, 4, 5],
    [5, 7, 4, 6],
    [6, 7, 3, 4]
  ];
  const tensor0 = tensor2d(array).expandDims(0);
  const tensor1 = tensor2d(array.reverse()).expandDims(0);

  const labels0 = [tensor1d([0.01]), tensor1d([0.02]), tensor1d([0.03])];
  const labels1 = [tensor1d([0.1]), tensor1d([0.09]), tensor1d([0.08])];
  const ds1 = data.array([tensor0, tensor1]);
  const ds2 = data.array([labels0, labels1]);
  const ds3 = data.zip([ds1, ds2]);

  const dataSet = ds3.map((x) => {
    return { xs: x[0], ys: x[1] };
  });

  const history = await tfModel.fitDataset(dataSet, {
    epochs: 20,
    verbose: 0,
    validationData: dataSet
  });
  console.log(history);
  const predictions = await tfModel.predict(tensor1);
  // @ts-ignore
  predictions.forEach((tensor) => {
    tensor.print();
  });
})();
```
https://codesandbox.io/s/typescript-node-tfjs-dunsw?file=/src/index.ts:2214-2224

**Other info / logs** 
Example output from codesanbox.io, see `val_y3_acc` as it somehow doesn't show correct value, but the actual predictions are good.

```
History {
  validationData: null,
  params: {
    epochs: 15,
    initialEpoch: null,
    samples: null,
    steps: 2,
    batchSize: null,
    verbose: 0,
    doValidation: true,
    metrics: [
      'loss',        'y1_loss',
      'y2_loss',     'y3_loss',
      'y1_acc',      'y2_acc',
      'y3_acc',      'val_loss',
      'val_y1_loss', 'val_y2_loss',
      'val_y3_loss', 'val_y1_acc',
      'val_y2_acc',  'val_y3_acc'
    ]
  },
  epoch: [
     0,  1,  2, 3,  4,  5,
     6,  7,  8, 9, 10, 11,
    12, 13, 14
  ],
  history: {
    val_loss: [
      0.06740279495716095,
[... omitted for readability],
      0.00021780715906061232
    ],
    val_y1_loss: [
      0.08847129344940186,
[... omitted for readability],
      0.0015886977780610323
    ],
    val_y2_loss: [
      0.10580939054489136,
[... omitted for readability],
      0.003424264956265688
    ],
    val_y3_loss: [
      0.06740279495716095,
[... omitted for readability],
      0.00021780715906061232
    ],
    val_y1_acc: [
      0.021068500354886055,
      0.010667561553418636,
      0.02188786491751671,
      0.022847648710012436,
      0.013655520044267178,
      0.004707253538072109,
      0.0018774932250380516,
      0.0036717751063406467,
      0.005106504075229168,
      0.0037899017333984375,
      0.0013052645372226834,
      0.000026645591788110323,
      0.0004705819010268897,
      0.001332651823759079,
      0.0013708906481042504
    ],
    val_y2_acc: [
      0.01733810268342495,
      0.0026979746762663126,
      0.016361301764845848,
      0.020113449543714523,
      0.01138104498386383,
      0.002396896481513977,
      0.0001758798462105915,
      0.003242085687816143,
      0.005758211947977543,
      0.004670063033699989,
      0.0017483194824308157,
      0.00013128435239195824,
      0.0006708669243380427,
      0.001789348665624857,
      0.0018355674110352993
    ],
    val_y3_acc: [
      0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0,
      0, 0, 0
    ],
    loss: [
      0.34463465213775635,
[... omitted for readability],
      0.003640228882431984
    ],
    y1_loss: [
      0.1969234198331833,
[... omitted for readability],
      0.00024854816729202867
    ],
    y2_loss: [
      0.07084406167268753,
[... omitted for readability],,
      0.0014955892693251371
    ],
    y3_loss: [
      0.07686717063188553,
[... omitted for readability],
      0.0018960912711918354
    ],
    y1_acc: [
      0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0,
      0, 0, 0
    ],
    y2_acc: [
      0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0,
      0, 0, 0
    ],
    y3_acc: [
      0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0,
      0, 0, 0
    ]
  }
}
```

**Comment by pyu10055** [comment]
@tuturis The `acc` metric counts how many time the output is the same as label throughout the training, it targets classification (with fixed label). In your case your output value is float,  the training results will never be exactly the same as the labels. 
So you will see `0` values in the metric.
You can try other metrics listed [here](https://js.tensorflow.org/api/latest/#Metrics)

**Comment by tuturis** [comment]
> @tuturis The `acc` metric counts how many time the output is the same as label throughout the training, it targets classification (with fixed label). In your case your output value is float,  the training results will never be exactly the same as the labels. 
> So you will see `0` values in the metric.
> You can try other metrics listed [here](https://js.tensorflow.org/api/latest/#Metrics) 

Thanks for the reply, noticed it was an regression problem and accuracy only partially makes sense for it. But I think there's still a bug because some of the accuracy metrics still report some values and only the third one is faulty. I would still consider this a bug.
